gb-cpu/src/memory/zone/rom/mod.rs

```diff
@@ -0,0 +1,41 @@
+mod controllers;
+mod mbc;
+
+use std::fs::File;
+use std::io::prelude::*;
+
+use controllers::romonly::*;
+use mbc::*;
+
+#[derive(Debug)]
+pub struct Rom {
```

Comment:
It will only dispatch to the right MBC, the MBC should do their own heavy-lifting

Reply:
So what it'll change from implementing directly `FileOperation` (what I'm doing currently) for each MBC controllers ?

Reply:
`FileOperation` is inspired from [linux file_operation](https://github.com/Rust-for-Linux/linux/blob/rust/rust/kernel/file_operations.rs#L607)

Reply:
I don't think we will have a lot of conflicts with the std `Read` and `Write` trait, but I am not sure why we should split `FileOperation` into two separate traits ?

Reply:
I actually had the conflict in the Rom, that's how i saw it.

But read and write (or get/ set) have slightly different needs, having them paired make implementing the trait way harder. Especialy since sometime write returns a result, sometimes it doesn't, and the type that is written can be different of the output.

Reply:
And some implementation only need the Read, like RomOnly/NoMbc

Reply:
I concerned about separating the `read/write` into their own trait will add a lot of complexity to the project since will have to handle the case where a controller implement the write trait and not the another one

Reply:
Oh then we just need to have write depend on Read

`trait Write<T>: Read`

It makes sense conceptualy: You can read without the ability to write, but if you write you can read too